python_modules/dagster/dagster/_core/definitions/asset_check_factories/freshness_checks/sensor.py

```diff
@@ -162,42 +162,87 @@ def freshness_checks_get_evaluations_iter(
             yield check_key, True
             continue
 
-        # Case 2: The check is currently evaluating. We shouldn't kick off another evaluation until it's done.
-        if (
-            summary_record.last_check_execution_record.status
-            == AssetCheckExecutionRecordStatus.PLANNED
-        ):
-            context.log.info(
-                f"Freshness check on asset {check_key.asset_key.to_user_string()} is in the planned state, indicating it is currently evaluating. Skipping..."
+        # Case 2: The check is currently evaluating and has never previously evaluated. We shouldn't kick off another evaluation until it's done.
+        if summary_record.last_completed_check_execution_record is None:
+            check.invariant(
+                summary_record.last_check_execution_record.status
+                == AssetCheckExecutionRecordStatus.PLANNED,
+                f"Unexpected status for check {check_key.to_user_string()}. The summary record indicates that the check has never completed, but the last check execution record is in terminal state {summary_record.last_check_execution_record.status}. This is likely a framework error, please report this to the Dagster maintainers.",
             )
-            yield check_key, False
-            continue
 
-        evaluation = check.not_none(
-            summary_record.last_check_execution_record.event
-        ).asset_check_evaluation
-        # Case 3: The check previously failed. We shouldn't kick off another evaluation until the asset has been updated.
```

Comment:
maybe we could grab the run associated and see if it is tagged as coming from this sensor as a way to avoid repeated submission but still submit a new one if there is a planned check and is overdue

Reply:
wait; but we can't tell if a planned check is overdue from just looking at the current evaluation right. Since we don't have access to the FRESH_UNTIL metadata.

Maybe we need to log "PREVIOUS_FRESH_UNTIL" on the check record or something?

Reply:
I think we want your thing either way though to avoid sending multiple requests